implementations/rust/kex/src/xx.rs

```diff
@@ -4,37 +4,35 @@ use crate::{CipherSuite, NewKeyExchanger, AES_GCM_TAGSIZE};
 use ockam_vault::{
     error::{VaultFailError, VaultFailErrorKind},
     types::{
-        PublicKey, SecretKey, SecretKeyAttributes, SecretKeyContext, SecretKeyType,
-        SecretPersistenceType, SecretPurposeType,
+        PublicKey, SecretKey, SecretKeyAttributes, SecretKeyType, SecretPersistenceType,
+        SecretPurposeType,
     },
-    DynVault,
+    DynVault, Secret,
 };
 use std::sync::{Arc, Mutex};
 use zeroize::Zeroize;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Debug)]
 struct KeyPair {
     public_key: PublicKey,
-    secret_handle: SecretKeyContext,
+    secret_handle: Box<dyn Secret>,
 }
 
 /// Represents the XX Handshake]
 struct SymmetricState {
     cipher_suite: CipherSuite,
-    static_key_pair: Option<KeyPair>,
+    identity_key: Option<Arc<Box<dyn Secret>>>,
```

Comment:
Arc is needed to prevent Secret Copy/Clone for long-term secrets. It looks like for ephemeral secrets it's possible to not use Copy/Clone even without Arc, since they have single owner

Reply:
How?

Reply:
If you cannot copy/clone identity key but want to use it from different places/threads you can achieve that by wrapping it inside Arc.
With ephemeral keys it's simpler, cause you just don't use it from different places